### DISCUSSION TITLE
Error after every 10 minutes org.springframework.data.redis.RedisConnectionFailureException: java.net.SocketException: Connection reset

### DISCUSSION THREAD

**Discussion by Shubhamlmp** [xetorthio/jedis#3981] [Q&A]
I'm getting below exception after every 10 mins
org.springframework.data.redis.RedisConnectionFailureException: java.net.SocketTimeoutException: Read timed out
redis.clients.jedis.exceptions.JedisConnectionException: java.net.SocketTimeoutException: Read timed out

```org.springframework.data.redis.RedisConnectionFailureException: java.net.SocketException: Connection reset
\tat org.springframework.data.redis.connection.jedis.JedisExceptionConverter.convert(JedisExceptionConverter.java:67)
\tat org.springframework.data.redis.connection.jedis.JedisExceptionConverter.convert(JedisExceptionConverter.java:42)
\tat org.springframework.data.redis.PassThroughExceptionTranslationStrategy.translate(PassThroughExceptionTranslationStrategy.java:40)
\tat org.springframework.data.redis.FallbackExceptionTranslationStrategy.translate(FallbackExceptionTranslationStrategy.java:38)
\tat org.springframework.data.redis.connection.jedis.JedisConnection.convertJedisAccessException(JedisConnection.java:212)
\tat org.springframework.data.redis.connection.jedis.JedisConnection.doWithJedis(JedisConnection.java:696)
\tat org.springframework.data.redis.connection.jedis.JedisConnection.doInvoke(JedisConnection.java:185)
\tat org.springframework.data.redis.connection.jedis.JedisConnection.lambda$new$0(JedisConnection.java:86)
\tat org.springframework.data.redis.connection.jedis.JedisInvoker$Synchronizer.invoke(JedisInvoker.java:1023)
\tat org.springframework.data.redis.connection.jedis.JedisInvoker.just(JedisInvoker.java:118)
\tat org.springframework.data.redis.connection.jedis.JedisStringCommands.get(JedisStringCommands.java:53)
\tat org.springframework.data.redis.connection.DefaultedRedisConnection.get(DefaultedRedisConnection.java:284)
\tat org.springframework.data.redis.core.DefaultValueOperations$1.inRedis(DefaultValueOperations.java:54)
\tat org.springframework.data.redis.core.AbstractOperations$ValueDeserializingRedisCallback.doInRedis(AbstractOperations.java:61)
\tat org.springframework.data.redis.core.RedisTemplate.execute(RedisTemplate.java:406)
\tat org.springframework.data.redis.core.RedisTemplate.execute(RedisTemplate.java:373)
\tat org.springframework.data.redis.core.AbstractOperations.execute(AbstractOperations.java:97)
\tat org.springframework.data.redis.core.DefaultValueOperations.get(DefaultValueOperations.java:50)
```
I'm not getting solution for this anywhere and I don't know why this is happening.
Thank You!

**Comment by sazzad16** [comment]
What is the `timeout` in your Redis server?

**Reply by Shubhamlmp** [reply]
> What is the `timeout` in your Redis server?
    Previously read timeout = 3 sec, connect timeout = 3 sec;
    But I increased from3 to 5 second, still same issue

**Reply by sazzad16** [reply]
I meant "in" your Redis server.
To ask it differently, what is the reply of `configGet("timeout")`?

**Reply by Shubhamlmp** [reply]
Sorry, In my redis server timeout is 0.
```
# redis-cli CONFIG GET TIMEOUT
1) "TIMEOUT"
2) "0"
```

**Reply by sazzad16** [reply]
The config seems fine. This is now difficult to guess the reason. Some of other possible reasons could be:

- Socket was closed before the operation which, I assume, is unlikely.
- Are you on Windows? If so, [check this](https://stackoverflow.com/a/4300803/4021802).

> Windows, but not Unix, Linux etc., resets a connection if a process exists without closing a socket. The other OSes close it properly.

**Reply by Shubhamlmp** [reply]
No, I'm on linux.

**Comment by sazzad16** [comment]
I now see you mentioned two different errors. For `java.net.SocketTimeoutException: Read timed out` in non-blocking commands, our usual suspect is network instability.

**Reply by Shubhamlmp** [reply]
But I'm not getting the point why this is happening in one service, why not in others.
I'm thinking of some issue with redis configuration itself in my service.

**Reply by Shubhamlmp** [reply]
@sazzad16 Is this issue related to pooling config like min and max Idle?

**Reply by sazzad16** [reply]
@Shubhamlmp I don't think so.
